src/host/path_normalize.c

```diff
@@ -8,128 +8,124 @@
 #include <ctype.h>
 #include <string.h>
 
-// isspace custom version (visual c++'s one doesn't like utf8 characters)
-static int is_space(char c)
-{
-	return  (c >= 9 && c <= 13) || c == 32;
-}
+#define	IS_SEP(__c)			((__c) == '/' || (__c) == '\\')
 
-static void* normalize_substring(const char* str, const char* endPtr, char* writePtr) {
-	const char* const source = str;
-	const char* const writeBegin = writePtr;
-	const char* ptr;
-	char last = 0;
-	char ch;
+#define IS_UPPER_ALPHA(__c)	((__c) >= 'A' && (__c) <= 'Z')
+#define IS_LOWER_ALPHA(__c)	((__c) >= 'a' && (__c) <= 'z')
+#define IS_ALPHA(__c)		(IS_UPPER_ALPHA(__c) || IS_LOWER_ALPHA(__c))
 
-	while (str != endPtr) {
-		ch = (*str);
+#define IS_SPACE(__c)		((__c >= '\t' && __c <= '\r') || __c == ' ')
 
-		/* make sure we're using '/' for all separators */
-		if (ch == '\\') {
-			ch = '/';
-		}
+static void* normalize_substring(const char* srcPtr, const char* srcEnd, char* dstPtr) {
 
-		/* filter out .. except when it's part of the file or folder name */
-		if (ch == '.' && last == '.' && *(str - 2) == '/' && (*(str + 1) == '/' || str + 1 == endPtr)) {
-			last = 0;
-
-			ptr = writePtr - 3;
-			while (ptr >= writeBegin) {
-				/* break on '/' except when it's '/../' */
-				if (ptr[0] == '/' && !(ptr[1] == '.' && ptr[2] == '.' && ptr[3] == '/')) {
-					writePtr -= writePtr - ptr;
-
-					/* special fix for cases, when '..' is the last chars in path i.e. d:\game\.., this should be converted into d:\,
-					but without this case, it will be converted into d: */
-					if (writePtr - 1 >= writeBegin && *(writePtr - 1) == ':' && str + 1 == endPtr) {
-						++writePtr;
-					}
-					break;
-				}
-				--ptr;
-			}
+#define IS_END(__p)			(__p >= srcEnd || *__p == '\0')
+#define IS_SEP_OR_END(__p)	(IS_END(__p) || IS_SEP(*__p))
 
-			if (ptr < writeBegin) {
-				*(writePtr++) = ch;
-			}
+	// Handle Windows absolute paths
+	if (IS_ALPHA(srcPtr[0]) && srcPtr[1] == ':')
+	{
+		*(dstPtr++) = srcPtr[0];
+		*(dstPtr++) = ':';
+
+		srcPtr += 2;
+	}
 
-			++str;
-			continue;
+	// Handle path starting with a sep (C:/ or /)
+	if (IS_SEP(*srcPtr))
+	{
+		++srcPtr;
+		*(dstPtr++) = '/';
+		// Handle path starting with //
+		if (IS_SEP(*srcPtr))
+		{
+			++srcPtr;
+			*(dstPtr++) = '/';
 		}
+	}
 
-		/* filter out /./ */
-		if (ch == '/' && last == '.') {
-			ptr = str - 2;
-			if (*ptr == '/') {	// there is no need to check whether ptr >= source since all the leading ./ will be skipped in path_normalize
-				if (ptr - 1 < source || *(ptr - 1) != ':') {
-					--writePtr;
-				}
+	const char * const dstRoot = dstPtr;
+	unsigned int folderDepth = 0;
```

Comment:
These probably need to be at the top of the function - some people build Premake with compilers that don't support this. I can't recall what, but we've had issues opened about it before - it might be older versions of VS.

Reply:
The issue I was thinking about is #696, so VS2010 will most like have the issue if it's a problem in VS2012.

Reply:
Indeed, this is an issue with C89 standard... I will fix that whenever I have some time.

There are other part of the code that have this issue, maybe set appveyor to build with vs2012 if possible?

Reply:
Oh, if it's elsewhere, let's just merge this! Thanks for looking into it!